bootstrap_conan_utils.py

```diff
@@ -0,0 +1,565 @@
+#        "compiler.threads" : "posix",
+#        "compiler.exception" : "seh",
+#        "arch" : "x86_64",
+#        "compiler" : "gcc",
+#        "build_type" : "Release",
+#        "compiler.version" : "11.2"
+#      },
+#  (...)
+#
+#  ; list revisions of a package for a specific recipe revision
+#  curl -v 'http://conan.jitx.com:8081/artifactory/api/conan/conan-local/v2/conans/pcre/8.45/_/_/revisions/125d5f684fea10391ff4cbcd809a5c74/packages/6f7dea16aa97d9ca0f6b67b413781234ab190708/revisions'
+#
+# --- Use hash to get download url
+#  curl $APIBASE/conans/pcre/8.45/_/_/packages/139391a944851d9dacf1138cff94b3320d5775dd/download_urls
+#  {
+#    "conaninfo.txt" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conaninfo.txt",
+#    "conan_package.tgz" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conan_package.tgz",
+#    "conanmanifest.txt" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conanmanifest.txt"
+#  }
+#
+# --- Use the url for "conan_package.tgz"
+#  http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conan_package.tgz
+
+
+def urlenc(s: str) -> str:
+    """Urlencode the given string"""
+    return urllib.parse.quote_plus(s)
+
+def conan_search_package_name(package_name: str, **kwargs) -> json:
+    """
+    Search for the given package name on the conan server
+
+    Parameters:
+      package_name: str The package name to find.  Only the name, no version components.
+    kwargs:
+      repourl: str The repo to search.  Optional, Default DEFAULT_CONAN_URL.
+
+    Returns: json results
+
+    throws Exception on invalid json returned
+    """
+    repourl = kwargs["repourl"] if "repourl" in kwargs else DEFAULT_CONAN_URL
+    queryurl = f"{repourl}/v2/conans/search"
+    headers = {"Content-Type": "application/json"}
+    params = {"q": package_name}
+
+    response = requests.get(queryurl, headers=headers, params=params)
+    return json.loads(response.text)
+
+
+def conan_get_recipe_revisions(package_name: str, package_version: str, **kwargs) -> dict:
+    """
+    Search for the available recipe revisions for the given package name on the conan server
+
+    Parameters:
+        package_name
+        package_version
+    kwargs:
+      repourl: str The repo to search.  Optional, Default DEFAULT_CONAN_URL.
+
+    throws Exception on invalid json returned
+    """
+    repourl = kwargs["repourl"] if "repourl" in kwargs else DEFAULT_CONAN_URL
```

Comment:
I would suggest making this is a function:

```python

def repourl(self, kwargs):
   return kwargs["repourl"] if "repourl" in kwargs else DEFAULT_CONAN_URL

...
repourl = self.repourl(kwargs)
```

Otherwise, you are repeating this a lot and this seems like the kind of thing we might want to expand in the future.